src/pages/Donate.jsx

```diff
@@ -1,101 +1,62 @@
-            };
-            script.onerror = () => {
-                resolve(false);
-            };
-            document.body.appendChild(script);
-        });
-    };
-
-    useEffect(() => {
-        loadScript("https://checkout.razorpay.com/v1/checkout.js");
+  document.title = "Milan | Donate the needy";
+
+  // Authentication state check !
+  const AuthState = () => {
+    const [login, setLogin] = useState(
+      Cookies.get("token") || Cookies.get("user")
+    );
+    return login;
+  };
+
+  // Assigning an Id for restricting multiple toast display
+  const toastId = "loginBeforeDonating";
+  if (!AuthState()) {
+    toast("🌈 Please login before supporting the clubs", {
+      toastId: toastId,
+      position: "top-right",
+      autoClose: 1000,
+      hideProgressBar: false,
+      closeOnClick: true,
+      pauseOnHover: true,
+      draggable: true,
+      progress: undefined,
+      closeButton: false,
     });
-
-
-    return (
-        <>
-
-            <Navbar />
-
-            <ToastContainer
-                position="top-right"
-                autoClose={1000}
-                hideProgressBar={false}
-                newestOnTop={false}
-                closeOnClick
-                rtl={false}
-                pauseOnFocusLoss
-                draggable
-                pauseOnHover
-                closeButton={false}
-            />
-
-            <div id="donate_banner"
-                className="container"
-            >
-
-                <div id="donateCol2">
-                    <img src={donate_image1} alt="" className="donate_img" />
-                </div>
-
-
-                <div
-                    id="donatecol_1"
-                    className="d-flex flex-column justify-content-center align-items-start me-5"
-                >
-                    <h1 className="mb-4">Yes, you help live !!</h1>
-                    <p>Donations does play an important part as our annual funds, donations from your ends helps thousands of unfortunate people live their lives. </p>{" "}
-
-                    <p>Choose any club, donate whatever you want, even 5 rupees helps !</p>
-
-
-                </div>
-
-            </div>
-
-
-            <hr className="container" />
-
-            <div className="cards justify-content-center">
-                {clubData.map((club) => {
-                    return <SingleClub key={club._id} club={club} />;
-                })}
-            </div>
-            <Footer />
-        </>
-    )
-}
-
-export default Donate
\ No newline at end of file
+  }
+
+  return (
+    <>
+      <ToastContainer
+        limit={1}
+        position="top-right"
+        autoClose={1000}
+        hideProgressBar={false}
+        newestOnTop={false}
+        closeOnClick
+        rtl={false}
+        pauseOnFocusLoss
+        draggable
+        pauseOnHover
+        closeButton={false}
+      />
+      {AuthState() && <Navbar />}
+      {AuthState() ? <DonateUs /> : <UserLogin />}
```

Comment:
What is happening is , before even seeing the toast we are getting redirected to UserLogin 
In short, we are not seeing the DonateUs  component - as a result, the toast doesnot works